File: src/main/java/cz/solutions/cockroach/DateInterval.java
package cz.solutions.cockroach;

import lombok.Value;
import org.joda.time.LocalDate;

@Value
public class DateInterval {
    private final LocalDate startInclusive;
    private final LocalDate endExclusive;

    public static DateInterval year(int year) {
        return new DateInterval(new LocalDate(year, 1, 1), new LocalDate(year + 1, 1, 1));
    }

    public boolean contains(LocalDate date) {
        return startInclusive.equals(date) || (startInclusive.isBefore(date) && date.isBefore(endExclusive));
    }
}

File: src/main/kotlin/cz/solutions/cockroach/DateInterval.kt
package cz.solutions.cockroach

import org.joda.time.LocalDate

data class DateInterval(val startInclusive: LocalDate, val endExclusive: LocalDate) {

    companion object {
        fun year(year: Int): DateInterval {
            return DateInterval(LocalDate(year, 1, 1), LocalDate(year + 1, 1, 1))
        }
    }

    fun contains(date: LocalDate): Boolean {
        return startInclusive == date || (startInclusive.isBefore(date) && date.isBefore(endExclusive))
    }
}